spring-kafka/src/main/java/org/springframework/kafka/support/serializer/ErrorHandlingDeserializer2.java

```diff
@@ -100,14 +96,18 @@ public ErrorHandlingDeserializer2(Deserializer<T> delegate) {
 	/**
 	 * Provides an alternative supplying mechanism when deserialization fails.
 	 * @param failedDeserializationFunction BiFunction type param to he invoked.
-	 * @since 1.3
+	 * @since 2.3
 	 * @deprecated in favor of {@link ErrorHandlingDeserializer2#setFailedDeserializationFunction(Function)}.
 	 */
-	@Deprecated
+	@Deprecated(since = "2.8")
```

Comment:
Wait. How does it work? I don't see any `since` in the `java.lang.Deprecated`.
I mean something like this in Javadocs:
```
* @deprecated since 2.3 in favor of #setFailedDeserializationFunction ...
```

You can find such a sample throughout the project.

More over that `2.8` is still wrong...

Reply:
I works over java 9 but with the current project environment the check fails.